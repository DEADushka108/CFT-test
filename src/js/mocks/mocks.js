export * from './const/const';
export * from './posts/posts';
export * from './store/store';
export * from './users/users';
export * from './comments/comments';
