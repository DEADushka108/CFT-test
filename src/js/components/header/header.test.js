import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {store} from '../../mocks/mocks';
import Header from './header';

it(`Should Header component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <Header/>
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
