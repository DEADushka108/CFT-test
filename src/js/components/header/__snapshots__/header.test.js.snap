// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header component render correctly 1`] = `
<header
  className="header__PageHeader-q13gpr-0 iSXoXP"
>
  <div
    className="header__Wrapper-q13gpr-1 kBoTBU"
  >
    <a
      className="header__Logo-q13gpr-2 bcCSEi"
      href="/"
      onClick={[Function]}
    >
      Chirrup
    </a>
    <nav
      className="header__MainNav-q13gpr-3 LPJae"
    >
      <ul
        className="header__List-q13gpr-4 gxvFWj"
      >
        <li
          className="header__Item-q13gpr-5 fYnSgo"
        >
          <a
            className="header__NavLink-q13gpr-6 hrrukg"
            href="/"
            onClick={[Function]}
          >
            <svg
              className="header__Icon-q13gpr-7 clURuZ"
            >
              <use
                xlinkHref="#home-page"
              />
            </svg>
            <span
              className="header__NavText-q13gpr-8 ggugkA"
            >
              Main
            </span>
          </a>
        </li>
        <li
          className="header__Item-q13gpr-5 fYnSgo"
        >
          <a
            className="header__NavLink-q13gpr-6 hrrukg"
            href="/users"
            onClick={[Function]}
          >
            <svg
              className="header__Icon-q13gpr-7 clURuZ"
            >
              <use
                xlinkHref="#social-group"
              />
            </svg>
            <span
              className="header__NavText-q13gpr-8 ggugkA"
            >
              Users
            </span>
          </a>
        </li>
        <li
          className="header__Item-q13gpr-5 fYnSgo"
        >
          <a
            className="header__NavLink-q13gpr-6 hrrukg"
            href="/contacts"
            onClick={[Function]}
          >
            <svg
              className="header__Icon-q13gpr-7 clURuZ"
            >
              <use
                xlinkHref="#message"
              />
            </svg>
            <span
              className="header__NavText-q13gpr-8 ggugkA"
            >
              Contact us
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
